flow/src/main/scala/org/alephium/flow/mempool/MemPool.scala

```diff
@@ -45,16 +45,20 @@ class MemPool private (
 
   def size: Int = pools.sumBy(_.size)
 
-  def contains(index: ChainIndex, transaction: TransactionTemplate): Boolean =
-    readOnly {
-      contains(index, transaction.id)
-    }
+  def contains(index: ChainIndex, transaction: TransactionTemplate): Boolean = {
+    contains(index, transaction.id)
```

Comment:
why did you removed the `readOnly`?

Reply:
the `readOnly` is provided by the internal `contains`